YahtzeeSim/Hand.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace YahtzeeSim
+{
+    public class Hand
+    {
+        public static List<int> currentDice = new List<int> { };
+
+        private int totalDice = 5;
+        private static Random rnd;
+
+        public Hand()
+        {
+            rnd = new Random();
+
+            currentDice = new List<int> { };
+            for (int i = 0; i < totalDice; i++)
+            {
+                currentDice.Add(RollDie());
+            }
+            
+        }
+
+        private static int RollDie()
+        {
+            return rnd.Next(1, 7);
+        }
+
+        public void ReRollDice(List<int> diePositions)
+        {
+            foreach (int position in diePositions)
+            {
+                if (position > 0 && position <= totalDice)
+                {
+                    currentDice[position - 1] = RollDie();
+                }
+            }
+        }
+
+        public String GetHandString()
+        {
+            String hand = "";
+
+            foreach (int dice in currentDice)
+            {
+                hand += dice.ToString() + " ";

```

Comment:
in C#, adding strings is actually less memory efficient. You might like StringBuilder instead. 
Also something like `hand += $"{dice.ToString()} "` I think. Hopefully I got that syntax right, it's string interpolation